tests/test_examples.py

```diff
@@ -42,67 +45,86 @@
 Z3_RELATED_TESTS = ["r9900_redundant_condition.py", "r9901_impossible_condition.py"]
 
 
-def get_file_paths(paths: Union[str, list[str]]) -> list[str]:
+def get_file_paths(paths: Union[str, list[str]], z3_enabled: bool = False) -> list[str]:
     """
     Get all the Python files from the specified directories for testing. This will
     return the full file paths for each Python file, excluding those listed in IGNORED_TESTS.
     The file paths will have the directory path prefixed to the file name for each element.
     A list of all the file paths will be returned.
 
     :param paths: The path or list of paths to retrieve the Python files from.
+    :param z3_enabled: Option to enable z3 dependency.
     :return: A list of full file paths to the Python files.
     """
     test_files = []
 
     if isinstance(paths, str):
         paths = [paths]
 
-    # check if z3 dependency is available
-    z3_dependency_available = True
-    try:
-        import z3
-    except ImportError:
-        z3_dependency_available = False
+    if z3_enabled:
```

Comment:
It's good to add a `z3_enabled` parameter explicitly, but in this case there shouldn't be a try-except block. When `z3_enabled` we can just run `import z3`, and if there's an error this should raised to the user, since the test isn't working as intended.